qiskit/circuit/quantumcircuit.py

```diff
@@ -242,8 +240,13 @@ def __init__(
             "qiskit.circuit.controlflow.builder.ControlFlowBuilderBlock"
         ] = []
 
-        self.qregs: list[QuantumRegister] = []
-        self.cregs: list[ClassicalRegister] = []
+        # Skip initialization if a subclass (e.g. BlueprintCircuit) is overriding
+        # these via properties.
+        if not hasattr(type(self), "qregs"):
+            self.qregs: list[QuantumRegister] = []
+        if not hasattr(type(self), "cregs"):
+            self.cregs: list[ClassicalRegister] = []
```

Comment:
We'd talked about not needing to do this in `QuantumCircuit`, but having any extra stuff in `BlueprintCircuit`, since it's the problematic element here by subclassing a class that's really not subclass-safe.  Is there a reason we don't do that in this form

I'm particularly concerned with static analysis and code completions for users getting way worse here, because of this apparently-dynamic setting of various properties.  If there's going to be grossness, I'd really rather that it's in `BlueprintCircuit` than `QuantumCircuit`.

Reply:
Done in 94c333d.